app/src/main/java/org/musicbrainz/picard/barcodescanner/activities/PerformSearchActivity.kt

```diff
@@ -74,44 +74,30 @@ class PerformSearchActivity : BaseActivity() {
     }
 
     private suspend fun search() {
-        val releaseList: Array<ReleaseInfo>
+        val result: BarcodeReleaseResponse
         try {
-            releaseList = releaseLookup(mBarcode)
-        } catch (e: IOException) {
+            result = MusicBrainzClient().instance.lookupReleaseWithBarcode(mBarcode!!)
+        }
+        catch (e: Exception){
             Log.e(this.javaClass.name, e.message, e)
             showResultActivityWithError(e.message)
             return
         }
-
-        try {
-            sendToPicard(releaseList)
-            showResultActivity(releaseList)
-        } catch (e: IOException) {
+        try{
+            sendToPicard(result.releases)
+            showResultActivity(result.releases)
+        }
+        catch (e: Exception){
             Log.e(this.javaClass.name, e.message, e)
             configurePicard()
```

Comment:
Currently doesn't work as expected, because exceptions are catched in PicardClient.openRelease already. So either remove the exception handling from PicardClient.openRelease or catch it there, but handle the boolean return type in `sendToPicard` and pass it through to here, then:

```
if (!sendToPicard(result.releases) {
    configurePicard()
}
```

Reply:
Thanks for the hint. I think we have handled it now.